src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -252,16 +270,6 @@ private static SafeAccessTokenHandle DuplicateAccessToken(SafeAccessTokenHandle
 
         private void CreateFromToken(IntPtr userToken)
         {
-            if (userToken == IntPtr.Zero)
-                throw new ArgumentException(SR.Argument_TokenZero);
-
-            // Find out if the specified token is a valid.
-            uint dwLength = (uint)sizeof(uint);
-            bool result = Interop.Advapi32.GetTokenInformation(userToken, (uint)TokenInformationClass.TokenType,
-                                                          SafeLocalAllocHandle.InvalidHandle, 0, out dwLength);
-            if (Marshal.GetLastWin32Error() == Interop.Errors.ERROR_INVALID_HANDLE)
-                throw new ArgumentException(SR.Argument_InvalidImpersonationToken);
-
```

Comment:
What impact does moving this code have?  Is there a test that fails before the change and passes after and on .NET Framework (and can you add that to the PR)?

Reply:
The additional check makes it possible for an ArgumentException to be thrown in cases where GetTokenInformation fails due to invalid handle or handle that is not an access token. Mainly it's just about making the checks similar to those done in netfx. Added a test.